app/models/user.rb

```diff
@@ -59,6 +59,19 @@ def toggle_role
         self.role = "default"
       end 
       save
+    end
+
+    def top_5_id_quantity
+      OrderItem.joins(:order)
+      .where("status='complete' AND user_id=#{self.id}")
+      .select(:item_id, :quantity).group(:item_id)
+      .order('sum_quantity DESC').limit(5).sum(:quantity)
+    end
```

Comment:
I don't think the relationship make sense to you Nico. Order's belong to default users.

In order to find a _merchants_ top 5 items - you have to include the _items_ table somehow.
* So what I would do here is move this method to OrderItem or Item and join `orders` AND `items`

The way you're using it here won't work. I promise. You're finding the top five items for a _default user_.
* order_items don't have a connection to merchants directly - they have an `item_id`
  * the `item_id` has a `user_id` _this_ is the connection to a merchant

I hope that makes sense. You don't have to take my word for it. Ask someone else, this isn't working the way you think it is. Please change it.

Reply:
yeahh this is why I asked if the test would work as is last night

Reply:
I'm still unsure of how orders/order_items/users/merchants interact